dkg-gadget/src/worker.rs

```diff
@@ -1290,42 +1295,44 @@ where
 
 			let (chain_id_type, ..): (ChainIdType<ChainId>, DKGPayloadKey) = key.clone();
 			debug!(target: "dkg", "Got unsigned proposal with key = {:?}", &key);
-			let data = match proposal {
-				ProposalType::RefreshProposal { data } => {
-					let refresh_prop_data =
-						match dkg_runtime_primitives::RefreshProposal::decode(&mut &data[..]) {
-							Ok(res) => res,
-							Err(err) => {
-								error!(target: "dkg", "Error decoding RefreshProposal {:?}", err);
-								continue
-							},
-						};
-					let mut buf = Vec::new();
-					buf.extend_from_slice(&refresh_prop_data.nonce.to_be_bytes());
-					buf.extend_from_slice(&refresh_prop_data.pub_key[..]);
-					Self::pre_signing_proposal_handler(chain_id_type, buf.to_vec())
-				},
-				ProposalType::AnchorUpdate { data } =>
-					Self::pre_signing_proposal_handler(chain_id_type, data),
-				ProposalType::TokenAdd { data } =>
-					Self::pre_signing_proposal_handler(chain_id_type, data),
-				ProposalType::TokenRemove { data } =>
-					Self::pre_signing_proposal_handler(chain_id_type, data),
-				ProposalType::WrappingFeeUpdate { data } =>
-					Self::pre_signing_proposal_handler(chain_id_type, data),
-				ProposalType::EVMUnsigned { data } => data,
-				_ => continue,
-			};
 
-			if let Err(e) = rounds.vote(key.encode(), data, latest_block_num) {
-				error!(target: "dkg", "🕸️  error creating new vote: {}", e.to_string());
+			if let Proposal::Unsigned { kind, data } = proposal {
+				use ProposalKind::*;
```

Comment:
Nit: Move to top of file.

Reply:
I don't want to pollute the namespace. There could be similar names in the future.

Reply:
I think I side with @dutterbutter here @appcypher. I would still put the imports or usages at the top of the file. Otherwise you can use the entire module path.

Reply:
@drewstone Just to be clear. These are for enum variants, not module path. They are typically not put at the top.
I can still move it to the top if you want.